portal/static/js/src/modules/OrgTool.js

```diff
@@ -350,6 +353,299 @@ export default (function() { /*global i18next $ */
+            }
+            if (a.name > b.name) {
+                return -1;
+            }
+            return 0;
+        });
+        childOrgs.forEach(function(item) {
+            var parentId = (item.partOf.reference).split("/")[2];
+            if (parentId) {
+                if (parentOrgsToDraw.indexOf(self.getOrgName(parentId)) !== -1) {
+                    return true;
+                }
+                var parentState = getParentState(parentId, states);
+                contentHTML = `<div class="radio"><label class="indent"><input class="clinic" type="radio" id="${item.id}_org" value="${item.id}" state="${parentState}" name="organization" data-parent-name="${item.name}" data-parent-id="${parentId}">${i18next.t(item.name)}</label></div>`;
+                if ($("#" + parentState + "_container legend[orgId='" + parentId + "']").length > 0) {
+                    $("#" + parentState + "_container legend[orgId='" + parentId + "']").after(contentHTML);
+                } else {
+                    $("#" + parentState + "_container").append(contentHTML);
+                }
+            }
+        });
+        //var selectOptions = $("#stateSelector").sortOptions();
+        var selectOptions = $("#stateSelector option");
+        if (selectOptions.length > 0) {
+            var selectSortedOptions = $("#stateSelector").sortOptions();
+            if (selectSortedOptions && selectSortedOptions.length > 0) { //sorting the select options
+                $("#stateSelector").empty().append(selectSortedOptions)
+                    .append(`<option value="none">${i18next.t("Other")}</option>`)
+                    .prepend(`<option value="" selected>${i18next.t("Select")}</option>`)
+                    .val("");
+            }
+            $(".state-container, .clinic-prompt").hide();
+            setTimeout(function() { //case of pre-selected clinic, need to check if any clinic has prechecked
+                var o = $("#userOrgs input[name='organization']:checked");
+                if (o.length > 0 && parseInt(o.val()) !== 0) {
+                    o.closest(".state-container").show();
+                    $(".clinic-prompt").show();
+                }
+            }, 150);
+            $("#userOrgs input[name='organization']").each(function() {
+                if (parseInt($(this).val()) !== 0) {
+                    Consent.getDefaultModal(this);
+                }
+            });
+            self.onLoaded(subjectId, false);
+        } else { // if no states found, then need to draw the orgs UI
+            $("#userOrgs .selector-show").hide();
+            self.onLoaded(subjectId, true);
+            self.filterOrgs(self.getHereBelowOrgs());
+            self.morphPatientOrgs();
+            $(".noOrg-container, .noOrg-container *").show();
+        }
+        $("#clinics").attr("loaded", true);
+        callback();
+    };
+    OrgTool.prototype.handleOrgsEvent = function(userId, isConsentWithTopLevelOrg) {
```

Comment:
Would be really cool to see all the org code as components :D

Reply:
yeah, I agree, there are others too that can be refactored into components.  These are a bit more involved changes though. I will do them when I have downtime again.